llvm/lib/Target/AMDGPU/SIPeepholeSDWA.cpp

```diff
@@ -1184,8 +1184,17 @@ bool SIPeepholeSDWA::convertToSDWA(MachineInstr &MI,
     if (PotentialMatches.count(Operand->getParentInst()) == 0)
       Converted |= Operand->convertToSDWA(*SDWAInst, TII);
   }
+
   if (Converted) {
     ConvertedInstructions.push_back(SDWAInst);
+    auto &MRI = SDWAInst->getParent()->getParent()->getRegInfo();
+    for (MachineOperand &MO : SDWAInst->uses()) {
+      if (!MO.isReg())
+        continue;
+    
+      if (!MRI.hasOneUse(MO.getReg()))
+        MRI.clearKillFlags(MO.getReg());
```

Comment:
You can just unconditionally clear the kill flags. They'll just get recomputed again later (and they've been soft deprecated for a long time)